go/libraries/doltcore/sql/sqltestutil/replacequeries.go

```diff
@@ -0,0 +1,321 @@
+			NewPeopleRow(9, "Jacqueline", "Bouvier", true, 80, 2),
+			NewPeopleRow(10, "Patty", "Bouvier", false, 40, 7),
+			NewPeopleRow(11, "Selma", "Bouvier", false, 40, 7),
+		),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind,
+			"is_married", types.BoolKind, "age", types.IntKind, "rating", types.FloatKind),
+	},
+	{
+		Name:         "replace partial columns multiple rows null pk",
+		ReplaceQuery: "replace into people (id, first, last) values (0, 'Bart', 'Simpson'), (1, 'Homer', null)",
+		ExpectedErr:  "column <last> received nil but is non-nullable",
+	},
+	{
+		Name:           "replace partial columns multiple rows duplicate",
+		ReplaceQuery:   "replace into people (id, first, last) values (2, 'Bart', 'Simpson'), (2, 'Bart', 'Simpson')",
+		SelectQuery:    "select id, first, last from people where id = 2",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(2), types.String("Bart"), types.String("Simpson"))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind),
+	},
+	{
+		Name: "replace partial columns existing pk",
+		AdditionalSetup: CreateTableFn("temppeople",
+			NewSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind, "num", types.IntKind),
+			NewRow(types.Int(2), types.String("Bart"), types.String("Simpson"), types.Int(44))),
+		ReplaceQuery:   "replace into temppeople (id, first, last, num) values (2, 'Bart', 'Simpson', 88)",
+		SelectQuery:    "select id, first, last, num from temppeople where id = 2",
+		ExpectedRows:   Rs(NewResultSetRow(types.Int(2), types.String("Bart"), types.String("Simpson"), types.Int(88))),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind, "num", types.IntKind),
+	},
+	{
+		Name: "replace partial columns multiple rows replace existing pk",
+		ReplaceQuery: `replace into people (id, first, last, is_married, age, rating) values
+					(0, "Homer", "Simpson", true, 45, 100),
+					(8, "Milhouse", "Van Houten", false, 8, 100)`,
+		SelectQuery: "select id, first, last, is_married, age, rating from people where rating = 100 order by id",
+		ExpectedRows: CompressRows(PeopleTestSchema,
+			NewPeopleRow(0, "Homer", "Simpson", true, 45, 100),
+			NewPeopleRow(8, "Milhouse", "Van Houten", false, 8, 100),
+		),
+		ExpectedSchema: NewResultSetSchema("id", types.IntKind, "first", types.StringKind, "last", types.StringKind,
+			"is_married", types.BoolKind, "age", types.IntKind, "rating", types.FloatKind),
+		SkipOnSqlEngine: true,
+	},
+	{
+		Name: "replace partial columns multiple rows null pk",
+		ReplaceQuery: `replace into people (id, first, last, is_married, age, rating) values
+					(0, "Homer", "Simpson", true, 45, 100),
+					(8, "Milhouse", "Van Houten", false, 8, 3.5),
+					(7, "Maggie", null, false, 1, 5.1)`,
+		ExpectedErr:     "Constraint failed for column 'last': Not null",
+		SkipOnSqlEngine: true,
```

Comment:
Should work with your recent change, right?

Reply:
It indeed does, I meant to remove these skips but I guess I forgot to.

Reply:
And I forgot to remove them from the Insert test file too lol